instances/devhub.near/widget/devhub/entity/proposal/Feed.jsx

```diff
@@ -1,7 +1,49 @@
-const { href } = VM.require("${REPL_DEVHUB}/widget/core.lib.url");
+const { href } = VM.require("${REPL_DEVHUB}/widget/core.lib.url") || {
+  href: () => {},
+};
+
+const instance = props.instance ?? "";
+const availableCategoryOptions = props.availableCategoryOptions ?? [];
+const announcement = props.announcement ?? <></>;
+
+let contract = "";
+let rfpFeedIndexerQueryName = "";
+let proposalFeedIndexerQueryName = "";
+let indexerHasuraRole = "";
+let isDevhub = false;
+let isInfra = false;
+let isEvents = false;
+
+switch (instance) {
+  case "infrastructure-committee.near": {
```

Comment:
Yeah, since this widget is defined in devhub.near and it's configuration is dependent on the instance that is using it, this will need to be passed via props and cannot be handled via a trick with aliases/replacements.

Although I would recommend, rather than passing props.instance here, and then doing switch case for query name, hasura role, query variables, filters, etc.... it may be easier and clearer to handle all unique logic in the calling component and pass it as props. I think state updates may perform better as well, since data query is happening in this widget too.

For example, props could be like:

```
{
  "contract": "${config_account}", // since your defining these props in app that uses it
  "getProposal": (proposalId) => {}, // optionally, instead of passing contract,
  "announcement": () => <>...</>,
  "additionalDetails": ({ proposal, rfp }) => <>...</>, // to replace 185-227
  "query": {
    indexerName: "...", // indexer name associated with app
    whereClause: (state) => {}
    handleData: (data) => {} // to replace lines 415 - 433 handleData(result.body.data).then(...)
    hasuraRole: "..."
  },
  "categories": [...],
}
```